### PR TITLE
[WIP] Create a node terminal even if there is no connectable.

### PR DESCRIPTION
**Pull request by mathbagu**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*
#490 

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
A NodeTerminal instance is created for each vertex in the Node/Breaker topology even if there is no connectable connected to this vertex.

**What is the current behavior?** *(You can also link to an open issue here)*
When two retained switches follow themselves, the Bus/Breaker view creates 3 buses. The bus in the middle has no NodeTerminal, so it is impossible to get/set voltage, angle connected and synchronous components.

**What is the new behavior (if this is a feature change)?**
A NodeTerminal is created for all valid vertices.

**Does this PR introduce a breaking change?** *(What changes might users need to make in their application due to this PR?)*
No

**Other information**:
As a NodeTerminal is created for each vertex, the memory consumption for a Network is higher. For a french grid, the number of NodeTerminal is 83 973 (instead of 46495) and the retained size of the network is 13% higher (191 Mo instead of 169 Mo).

I tried to be smarter, adding attributes in ConfiguredBusImpl to store voltage, angle connected and synchronous components. It's not working because there is no link between buses of the Bus/Breaker view and the Bus view.

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java [hunk 1]
```diff
@@ -68,13 +68,17 @@ public int getConnectedTerminalCount() {
...
+                .map(Function.identity());
     }
 
     @Override
     public Collection<TerminalExt> getTerminals() {
         checkValidity();
-        return Collections.unmodifiableCollection(terminals);
+        return terminals.stream()
+                .filter(t -> t.getConnectable() != null)
+                .collect(Collectors.toList());
```
Comment (sylvlecl) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java]: Should we reuse the getConnectedTerminalStream() method here ? Or maybe you intended to skip the "Function.identity()" step ?

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java [hunk 2]
```diff
@@ -68,13 +68,17 @@ public int getConnectedTerminalCount() {
     @Override
     public Stream<TerminalExt> getConnectedTerminalStream() {
         checkValidity();
-        return terminals.stream().map(Function.identity());
+        return terminals.stream()
+                .filter(t -> t.getConnectable() != null)
```
Comment (sylvlecl) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/CalculatedBusImpl.java]: I think the method getConnectedTerminalCount is not consistent anymore with it.

File: iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java [hunk 3]
```diff
@@ -565,13 +555,24 @@ private Integer getEdge(String switchId, boolean throwException) {
...
+                .filter(NodeBreakerVoltageLevel::isValid)
+                .map(Function.identity());
+    }
+
+    private static NodeTerminal getTerminal(UndirectedGraph<NodeTerminal, SwitchImpl> graph, int node) {
+        NodeTerminal terminal = graph.getVertexObject(node);
+        return isValid(terminal) ? terminal : null;
+    }
+
+    private static boolean isValid(NodeTerminal nodeTerminal) {
```
Comment (sylvlecl) [inline iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NodeBreakerVoltageLevel.java]: For this method and the getTerminal method, I think we should either rename them or explain in comments/javadoc the reason for considering terminals with no connectable "invalid".
Maybe we should document this design choice (having Terminals for all vertices, but not using them for most methods) in the header?

**Comment by geofjamg** [comment]
@mathbagu as #824 seems to be better and simpler, we can close this PR?

**Comment by geofjamg** [comment]
Solved by #824

**Pull request closed without merging**